frontend/src/DiagramEditor/DiagramEditor.jsx

```diff
@@ -49,7 +65,10 @@ export class DiagramEditor extends React.Component {
     window.addEventListener("mouseup", this.onWindowMouseUp);
     window.addEventListener("mousemove", this.onWindowMouseMove);
 
-    this.props.joinDiagram(this.props.match.params.diagramId);
+    this.generateAuthorId();
```

Comment:
I'm just wondering if this should be a global app thing or if this should be a per diagram thing? Our discussion earlier suggested that we would just set it once in a cookie and then reuse it... It's a change that can happen later, I just thought of it when I saw this. What are your thoughts?